File: p067.py
def p067(data):
    triangle = [[int(i) for i in line.split(" ")] for line in data]
    for i in xrange(len(triangle) - 2, -1, -1):
        for j in xrange(0, len(triangle[i])):
            triangle[i][j] += max(triangle[i+1][j:j+2])
    return triangle[0][0]

if __name__ == "__main__":
    from benchmark import benchmark
    benchmark(p067, open("data/p067.txt"))

File: p067.clj
(use '[euler :only [run]])

(defn p067 [file]
  (let [content  (slurp file)
        lines    (clojure.string/split content #"\n")
        to-int   #(Integer/parseInt %)
        triangle (map #(map to-int (clojure.string/split % #" ")) lines)]
    (first (reduce (fn [a b]
                     (map #(+ (nth b %) (apply max (subvec (vec a) % (+ 2 %))))
                          (range 0 (count b))))
                   (reverse triangle)))))


(run (p067 "data/p067.txt"))
